src/react/components/common/assetPreview/videoAsset.tsx

```diff
@@ -138,38 +153,97 @@ export class VideoAsset extends React.Component<IVideoAssetProps> {
     }
 
     /**
-     * Seeks the video to the timestamp specified in the component properties
+     * @name - Move to the next expected frame
+     * @description - Moves the videos current position forward one frame based on the current
+     * project settings for frame rate extraction
      */
-    private seekToTimestamp = () => {
-        if (this.props.timestamp > 0) {
-            this.videoPlayer.current.pause();
-            this.videoPlayer.current.seek(this.props.timestamp);
+    private moveNextExpectedFrame = () => {
+        // Seek forward from the current time to the next logical frame based on project settings
+        const frameSkipTime: number = (1 / this.props.project.videoSettings.frameExtractionRate);
+        const seekTime: number = (this.videoPlayer.current.getState().player.currentTime + frameSkipTime);
+        this.seekToTime(seekTime);
+    }
+
+    /**
+     * @name - Move to the previous expected frame
+     * @description - Moves the videos current position backward one frame based on the current
+     * project settings for frame rate extraction
+     */
+    private movePreviousExpectedFrame = () => {
+        // Seek backwards from the current time to the next logical frame based on project settings
+        const frameSkipTime: number = (1 / this.props.project.videoSettings.frameExtractionRate);
+        const seekTime: number = (this.videoPlayer.current.getState().player.currentTime - frameSkipTime);
+        this.seekToTime(seekTime);
+    }
+
+    /**
+     * @name - Move to nearest child frame
+     * @description - Move to the nearest tagged or visited frame from where the video's current
+     * position is, within a certain threshold
+     * @returns true if it moved position; false otherwise
+     */
+    private moveToNearestChildFrame(): boolean {
+        const maximumThreshold = 1; // one second threhsold
+        const timestamp = this.videoPlayer.current.getState().player.currentTime;
```

Comment:
Looks like we have quite a few places where we are calling `this.videoPlayer.current.getState()` now.  May we should make a tiny helper method that returns the current state mapped to our `IVideoPlayerState` interface.

Reply:
Updated